File: rest-service/src/main/java/mb/api/model/IOLongJobResponse.java
package mb.api.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import mb.api.model.blast.IOBlastConfig;
import mb.api.model.io.JsonKeys;

@JsonDeserialize(as = IOLongJobResponseImpl.class)
public interface IOLongJobResponse extends IOShortJobResponse
{
  @JsonProperty(JsonKeys.Config)
  IOBlastConfig getConfig();

  @JsonProperty(JsonKeys.Config)
  IOLongJobResponse setConfig(IOBlastConfig config);
}

File: rest-service/src/main/java/mb/api/model/IOLongJobResponse.kt
package mb.api.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.fasterxml.jackson.annotation.JsonProperty
import mb.api.model.blast.IOBlastConfig
import mb.api.model.io.JsonKeys
import mb.lib.model.JobStatus

@JsonInclude(NON_NULL)
class IOLongJobResponse(
  id: String? = null,
  description: String? = null,
  status: JobStatus? = null,
  created: String? = null,
  expires: String? = null,
  maxResultSize: Long? = null,
  parentJobs: Array<IOJobLink>? = null,
  childJobs: Array<IOJobLink>? = null,
  isPrimary: Boolean,
  site: String? = null,
  targets: Array<IOJobTarget>? = null,
  @JsonProperty(JsonKeys.Config) val config: IOBlastConfig
) : IOShortJobResponse(
  id, description, status, created, expires, maxResultSize, parentJobs,
  childJobs, isPrimary, site, targets
)